webui/react/src/components/kit/Spinner.tsx

```diff
@@ -12,20 +13,20 @@ interface PropsBase {
   tip?: React.ReactNode;
 }
 
-type Props<T> = (
-  | {
-      children?: React.ReactNode;
-      conditionalRender?: boolean;
-      data?: never;
-      spinning?: boolean;
-    }
-  | {
-      children: (data: T) => JSX.Element;
-      conditionalRender?: never;
-      data: Loadable<T>;
-      spinning?: never;
-    }
-) &
+type Props<T> = XOR<
```

Comment:
I don't realize understand what the `XOR` type does; it's not documented.

Reply:
It's a utility type that allows you to create mutually exclusive types where either the first or second set of props must be present, but not both. I got it from [here](https://stackoverflow.com/a/53229857). Basically, it does the same thing as the union that was there previously, but more cleanly. Of course, I forgot to actually finish cleaning it up until now, but it means that the props can look like this, without any `never`s in the mix:
```
type Props<T> = XOR<
  {
    children?: React.ReactNode;
    conditionalRender?: boolean;
    spinning?: boolean;
  },
  {
    children: (data: T) => JSX.Element;
    data: Loadable<T>;
  }
> &
  PropsBase;
```

Reply:
I added a link to the StackExchange answer next to the XOR definition.

Reply:
Makes sense. With the `never`s removed the function is clear.